Document that we accept dots in database names (#697)
neo4j/neo4j-documentation: cypher/cypher-docs/src/docs/dev/syntax/naming.asciidoc

[NOTE]
Non-alphabetic characters, including numbers, symbols and whitespace characters, *can* be used in names, but *must* be escaped using backticks.
For example: `++`^n`++`, `++`1first`++`, `++`$$n`++`, and `++`my variable has spaces`++`.

> Only database names are allowed to include dots without the need to escape the name.

Reviewer: ```suggestion
Database names are an exception and may include dots without the need for escaping.
```
